tests/Signal/ported.test.ts

```diff
@@ -0,0 +1,207 @@
+import {it, describe, expect, vi} from 'vitest';
+import {Signal} from '../../src';
+
+const watch = (cb) => {};
+
+describe('Ported signal tests', () => {
+  /**
+   * Url: https://github.com/vuejs/core/blob/main/packages/reactivity/__tests__/computed.spec.ts#L32
+   */
+  it('should return updated value', () => {
+    const s = new Signal.State<{foo?: number}>({});
+    const c = new Signal.Computed(() => s.get().foo);
+
+    expect(c.get()).toBe(undefined);
+    s.set({foo: 1});
+    expect(c.get()).toBe(1);
+  });
+
+  /**
+   * Url: https://github.com/vuejs/core/blob/main/packages/reactivity/__tests__/computed.spec.ts#L54
+   */
+  it('should compute lazily', () => {
+    const s = new Signal.State<{foo?: number}>({});
+    const getter = vi.fn(() => s.get().foo);
+    const c = new Signal.Computed(getter);
+
+    // lazy
+    expect(getter).not.toHaveBeenCalled();
+
+    expect(c.get()).toBe(undefined);
+    expect(getter).toHaveBeenCalledTimes(1);
+
+    // should not compute again
+    c.get();
+    expect(getter).toHaveBeenCalledTimes(1);
+
+    // should not compute until needed
+    s.set({foo: 1});
+    expect(getter).toHaveBeenCalledTimes(1);
+
+    // now it should compute
+    expect(c.get()).toBe(1);
+    expect(getter).toHaveBeenCalledTimes(2);
+
+    // should not compute again
+    c.get();
+    expect(getter).toHaveBeenCalledTimes(2);
+  });
+
+  /**
+   * Url: https://github.com/vuejs/core/blob/main/packages/reactivity/__tests__/computed.spec.ts#L189
+   */
+  it.skip('should invalidate before non-computed effects', () => {
+    let plusOneValues: number[] = [];
+    const n = new Signal.State(0);
+    const plusOne = new Signal.Computed(() => n.get() + 1);
+    watch(() => {
+      n.get();
+      plusOneValues.push(plusOne.get());
+    });
+
+    plusOne.get();
+    // mutate n
+    n.set(2);
+    // on the 2nd run, plusOne.value should have already updated.
+    expect(plusOneValues).toMatchObject([1, 2]);
+  });
+
+  /**
+   * Url: https://github.com/vuejs/core/blob/main/packages/reactivity/__tests__/computed.spec.ts#L488
+   */
+  it.skip('should work when chained(ref+computed)', () => {
+    const v = new Signal.State(0);
+    const c1 = new Signal.Computed(() => {
+      if (v.get() === 0) {
+        v.set(1);
+      }
+      return 'foo';
+    });
+    const c2 = new Signal.Computed(() => v.get() + c1.get());
+    expect(c2.get()).toBe('0foo');
+    expect(c2.get()).toBe('1foo');
+  });
+
+  /**
+   * Url: https://github.com/vuejs/core/blob/main/packages/reactivity/__tests__/computed.spec.ts#L502
+   */
+  it.skip('should trigger effect even computed already dirty', () => {
```

Comment:
why skipped?

Reply:
All of the tests in `ported.test.ts` with `.skip` fails so far. I haven't done so much research to why that is yet.